cluster-operator/src/main/java/io/strimzi/operator/cluster/operator/resource/KafkaRoller.java

```diff
@@ -454,9 +455,11 @@ private void restartIfNecessary(NodeRef nodeRef, RestartContext restartContext)
         // change and the desired roles still apply.
         boolean isBroker = Labels.booleanLabel(pod, Labels.STRIMZI_BROKER_ROLE_LABEL, nodeRef.broker());
         boolean isController = Labels.booleanLabel(pod, Labels.STRIMZI_CONTROLLER_ROLE_LABEL, nodeRef.controller());
+        // This is relevant when creating admin client for controllers
+        String currentVersion = pod.getMetadata().getAnnotations().get(Annotations.ANNO_STRIMZI_KAFKA_VERSION);
```

Comment:
Do we know that `pod` is not null here? (you can also use `Annotations.stringAnnotation(...)` to extract it with null checks, but the resource has to be non-null already)

Reply:
Yes on line 416, we check if pod is null and if it is, we return.